katdal/dataset.py

```diff
@@ -786,6 +786,15 @@ def select(self, **kwargs):
                         continue
                     target_keep |= (target_index_sensor == target_index)
                 self._time_keep &= target_keep
+            elif k == 'target_tags':
+                tags = v if is_iterable(v) else [v]
+                target_keep = np.zeros(len(self._time_keep), dtype=np.bool)
+                target_index_sensor = self.sensor.get('Observation/target_index')
+                for target_index in self.target_indices:
```

Comment:
I would not use `self.target_indices` inside `select`, since that attribute is selection-dependent and you will in effect be applying a selection on top of another selection. Rather work with `self.catalogue`, which is selection-independent.

Reply:
Good point, this would have caused a bug on second select...